scalyr_agent/__scalyr__.py

```diff
@@ -90,6 +91,55 @@
 __is_frozen__ = hasattr(sys, "frozen")
 
 
+# The constants for INSTALL_TYPE, a variable declared down below.
+PACKAGE_INSTALL = 1  # Indicates source code was installed via a package manager such as RPM or Windows executable.
+TARBALL_INSTALL = 2  # Indicates source code was installed via a tarball created by the build_package.py script.
+DEV_INSTALL = 3  # Indicates source code is running out of the original source tree, usually during dev testing.
+
+
+def read_install_info():
+    """
+    Read the 'install_info' file that has to be located near this file. In opposite return empty dict.
+    """
+    install_info_path = os.path.join(os.path.dirname(__file__), "install_info")
+    if not os.path.exists(install_info_path):
+        return {}
+
+    with open(install_info_path, "r") as fp:
+        return json.load(fp)
+
+
+__install_info__ = read_install_info()
+
+
+def __determine_install_type():
+    """Returns the type of install that was used for the source currently running.
+
+    @return: The install type, drawn from the constants above.
+    @rtype: int
+    """
+    # Determine which type of install this is by checking the 'install_info' file, that
+    # should be located near the this file.
+    install_type = __install_info__.get("install_type")
+
+    # If there's no  install_type field, then consider it as dev run.
+    if not install_type:
+        return DEV_INSTALL
+
+    if install_type in ["package"]:
+        return PACKAGE_INSTALL
+    elif install_type == "tar":
+        return TARBALL_INSTALL
+
+    raise ValueError(
+        "Unrecognized package type {} in the 'build_info' file".format(install_type)
```

Comment:
Perhaps also include actual value in the error message to make it easier / faster to troubleshoot?